packages/material-ui/src/MenuList/MenuList.d.ts

```diff
@@ -24,6 +24,7 @@ export interface MenuListProps extends StandardProps<ListProps, MenuListClassKey
    * If `true`, the menu items will not wrap focus.
    */
   disableListWrap?: boolean;
+  isSubMenu?: boolean;
```

Comment:
I suspect this is easy to forget. Can we create a context instead and the MenuList decides on itself whether it's a submenu or not?

Reply:
This isn't something that an implementer will ever pass. It's internally provided. I might have messed up how this is documented.

Reply:
You would need to add:
```js
  /**
   * @ignore
   */
```

Reply:
> This isn't something that an implementer will ever pass.

If it's part of the typings it's part of the public API. People also tend to inspect our source code and use it regardless of what we intended. So the better we isolate the safer we are when refactoring.

Reply:
I misspoke a slightly, there. I should have said it's not something an implementer _should_ ever pass, but point taken about users abusing what there is to abuse. 

I looked into a adding a context and I like the idea, but I'm not sure how to implement it correctly because of the recursive pattern in play. Since a root Menu is the same component as a SubMenu (for now), it seems like we'd be wrapping each layer in a new context of the same type and it would get very muddy if it even worked. 

Because the context is implemented via a JSX component wrapper, I'm not sure if we want to introduce some ternary logic to either wrap or not wrap with that component accordingly. Seems like it would be really messy/noisy.

Alternatively, I can just try to document it more correctly, so it just isn't listed in the public api docs?

Reply:
> Because the context is implemented via a JSX component wrapper, I'm not sure if we want to introduce some ternary logic to either wrap or not wrap with that component accordingly.

What about branching in the context value rather than branching at the component level:

```diff
-shouldHaveContext ? <Context value={contextValue}> : null
+<Context value={shouldHaveContext ? contextValue : null}>
```

Reply:
This is another weaker area for me, so I'm not sure, but it seems like you'd still end up with layered contexts, wouldn't you? Is that a problem? I'm not sure.

To illustrate, I'm worried that, in the example below, the root `Menu` will have a context that has useful things in it. Then, the `SubMenu` will also have another instance of the same Context but will be null/empty. Is that true? Is it a problem?  Remember that the `SubMenu` component is just a thin wrapper around the `Menu` component at this point, so both render essentially the same code. 

``` js
<Menu id="menu-1" {...props}>
  <MenuItem subMenu={(
    <SubMenu>
      <MenuItem>Something Else</MenuItem>
    </SubMenu>
  )}>
    Something
  </MenuItem>
</Menu>
```